llvm/lib/Target/AMDGPU/SOPInstructions.td

```diff
@@ -1160,6 +1158,9 @@ def S_SETREG_IMM32_B32_mode : S_SETREG_IMM32_B32_Pseudo {
 
 } // End Defs = [MODE], Uses = [MODE]
 
+def : GCNPat<(int_amdgcn_s_setreg (i32 timm:$simm16), i32:$sdst),
```

Comment:
Could define an SDNodeXForm for TimmLeaf and avoid splitting out the pattern from the instruction definition

Reply:
Done